### PR TITLE
Introduce new Featurizers packaging pipeline.

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**: 
This introduces a new Packaging pipeline for ORT build with Featurizers.
Featurizers are going to be removed from other pipelines and ORT-NIGHTLY feed
will only be pushed from this pipeline

**Motivation and Context**
As it stand now, Featurizers will not be a part of 1.2 release.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/nuget/templates/cpu-featurizers.yml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+  - task: PublishPipelineArtifact@0
+    displayName: 'Publish Pipeline NuGet Artifact'
+    inputs:
+      artifactName: 'drop-signed-nuget'
+      targetPath: '$(Build.ArtifactStagingDirectory)'
+   # XXX: Only windows now
+- template: test_win.yml
+
+- job: Publish_NuGet_Package_And_Report
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/nuget/templates/cpu-featurizers.yml]: We don't need to report anything from this pipeline to the dashboard.

File: tools/ci_build/github/azure-pipelines/nuget/templates/cpu-featurizers.yml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  workspace:
+    clean: all
+  variables:
+  - group: Dashboard_MySQL_Secret
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/nuget/templates/cpu-featurizers.yml]: We don't need to report anything from this pipeline to the dashboard like size of binary, code coverage numbers, etc.

File: tools/ci_build/github/azure-pipelines/nuget/templates/cpu-featurizers.yml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    demands: azureps
+  condition: and (succeeded(), and (${{ parameters.DoEsrp }}, eq(variables['Build.SourceBranch'], 'refs/heads/master')))
+  dependsOn:
+  - NuGet_Test_Win
+  steps:
+
+  - template: ../../templates/set-version-number-variables-step.yml
+
+    ## XXX: Only windows now
+  - template: upload-binary-sizes-from-featurizers-nuget-package.yml
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/nuget/templates/cpu-featurizers.yml]: You can skip this one.

File: tools/ci_build/github/azure-pipelines/nuget/templates/cpu-featurizers.yml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+     cd $(Build.SourcesDirectory)\csharp\src\Microsoft.ML.OnnxRuntime\bin\RelWithDebInfo\
+     ren *.nupkg win-arm64.zip
+     copy $(Build.SourcesDirectory)\csharp\src\Microsoft.ML.OnnxRuntime\bin\RelWithDebInfo\*zip $(Build.ArtifactStagingDirectory)
+     mkdir $(Build.ArtifactStagingDirectory)\testdata
+     copy $(Build.BinariesDirectory)\RelWithDebInfo\RelWithDebInfo\custom_op_library.* $(Build.ArtifactStagingDirectory)\testdata
+
+- template: ../../templates/win-ci-2019.yml
+  parameters:
+    AgentPool : 'Win-CPU-2019'
+    ArtifactName: 'drop-win-arm-zip'
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/nuget/templates/cpu-featurizers.yml]: Do you need the ARM build? If not, remove it for simplicity.

**Comment by pranavsharma** [comment]
Should we change the name of the package when pushing to the ORT-NIGHTLY feed so that existing customers of this feed don't see something awry?

**Comment by pranavsharma** [comment]
> Should we change the name of the package when pushing to the ORT-NIGHTLY feed so that existing customers of this feed don't see something awry?

LGTM except for this. I believe you'll check with David and get back.

**Comment by yuslepukhin** [comment]
I spoke with Harish and he asked not to change the name as it would involve more changes on their said which later would need to be reverted.

---
In reply to: [589920500](https://github.com/microsoft/onnxruntime/pull/3068#issuecomment-589920500) [](ancestors = 589920500)

**Pull request merged**